src/components/views/context_menus/RoomGeneralContextMenu.tsx

```diff
@@ -213,18 +217,33 @@ export const RoomGeneralContextMenu: React.FC<RoomGeneralContextMenuProps> = ({
     }
 
     const { level } = useUnreadNotifications(room);
-    const markAsReadOption: JSX.Element | null =
-        level > NotificationLevel.None ? (
-            <IconizedContextMenuCheckbox
-                onClick={() => {
-                    clearRoomNotification(room, cli);
-                    onFinished?.();
-                }}
-                active={false}
-                label={_t("room|context_menu|mark_read")}
-                iconClassName="mx_RoomGeneralContextMenu_iconMarkAsRead"
-            />
-        ) : null;
+    const markAsReadOption: JSX.Element | null = (() => {
+        if (level > NotificationLevel.None) {
+            return (
+                <IconizedContextMenuOption
+                    onClick={wrapHandler(() => {
+                        clearRoomNotification(room, cli);
+                        onFinished?.();
+                    }, onPostMarkAsReadClick)}
+                    label={_t("room|context_menu|mark_read")}
+                    iconClassName="mx_RoomGeneralContextMenu_iconMarkAsRead"
+                />
```

Comment:
can you walk me through why this is changing from `IconizedContextMenuCheckbox`, and what the effects are? And what does `wrapHandler` do and why do we need it now when we didn't before?

Reply:
As far as I can see, it being a checkbox must have been a c+p error. There's no reason for it to be a checkbox: it's, it's just a regular menu option. I think it just happened to work. `wrapHandler` is to add the call to onPostMarkAsReadClick (which triggers the analytics) so it's part of adding analytics.